### PR TITLE
ci: add validation action

### PR DESCRIPTION
**Pull request by satta**
This PR adds simple YAML syntax validation as a GitHub Action. It uses `yamllint` to check the syntax of the `index.yaml` file containing the index. Both merges to master and pull requests will be checked, and findings in the latter will be posted as comments to the respective PR.

Opinionated checks like maximum line length enforcement have been disabled.

### CODE REVIEW

File: index.yaml [hunk 1]
```diff
@@ -1,3 +1,5 @@
+---
+
```
Comment (jasonish) [inline index.yaml:2]: Can we remove the requirement for this with https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.document_start?

File: .github/workflows/yamllint.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+
+name: YAML Validation
+
+on:
+  push:
+    branches: [ master ]
```
Comment (jasonish) [inline .github/workflows/yamllint.yml:7]: On al pushes? I'd rather get the error on push to branch before push to master.

File: index.yaml (hunk 1 above)
Comment (satta) [reply index.yaml:2]: Yes, I never was a fan of that either.

File: .github/workflows/yamllint.yml (hunk 2 above)
Comment (satta) [reply .github/workflows/yamllint.yml:7]: Done. I think for such cases the PR check, which also provides errors in comments, is more useful.

**Comment by satta** [comment]
Next PR: #26

**Pull request closed without merging**